sensor/common/image/service_impl.go

```diff
@@ -55,7 +55,9 @@ func (s *serviceImpl) GetImage(ctx context.Context, req *sensor.GetImageRequest)
 
 	// Note: The Admission Controller does NOT know if the image is cluster-local,
 	// so we determine it here.
-	req.Image.IsClusterLocal = imageutil.IsInternalImage(req.GetImage().GetName())
+	// If Sensor's registry store has an entry for the given image's registry,
+	// it is considered cluster-local.
+	req.Image.IsClusterLocal = registry.Singleton().HasRegistryForImage(req.GetImage().GetName())
```

Comment:
Ideally, the registry should be a field in the `serviceImpl` rather than us invoking `registry.Singleton` here.

Reply:
done